src/redux/cartoSlice.js

```diff
@@ -130,6 +131,23 @@ export const createCartoSlice = (initialState) => {
         if (state.viewportFeatures[sourceId]) {
           delete state.viewportFeatures[sourceId];
         }
+      },
+      setWidgetLoaders: (state, action) => {
+        const { widgetId, isLoading, displayAllLoaders } = action.payload;
+
+        if (displayAllLoaders) {
+          state.widgetLoaders = Object.fromEntries(
+            Object.keys(state.widgetLoaders).map((key) => [key, true])
```

Comment:
key --> widgetId, right?

Reply:
Indeed! Are we overusing `widgetId` here?